google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -175,19 +177,68 @@ public Table create(TableInfo tableInfo, TableOption... options) {
 
   @Override
   public Job create(JobInfo jobInfo, JobOption... options) {
+    Supplier<JobId> idProvider =
+        new Supplier<JobId>() {
+          @Override
+          public JobId get() {
+            return JobId.of();
```

Comment:
Isn't there a way to pass in a prefix for the JobId, too? I recall that was a requirement from the BQ team. Or is there already a supplier for that?

Reply:
According to the docs, the recommended way is `JobId.of("my-prefix-" + UUID.randomUUID())`. Unfortunately it doesn't benefit from this change.

This is a good point, but I'm not sure what the right design is here.

Reply:
Perhaps let’s follow up in a separate issue/PR and add another static constructor to JobId with a prefix, maybe `JobId.withPrefix(String)`

Reply:
In current design, this would still be considered "provided JobId" though and we won't be able to call `get` if `create` fails. Are you OK with this?

If we want `JobId.withPrefix` to benefit from this, we'd actually need the `JobId` to know whether it was randomly generated.

Reply:
You're right. `withPrefix` wouldn't solve the problem. Rather, we'd want to be able to provide a prefix to the factory / supplier constructor.